src/services/findAllReferences.ts

```diff
@@ -388,17 +405,29 @@ namespace ts.FindAllReferences {
         return { ...entryToDocumentSpan(entry), ...(providePrefixAndSuffixText && getPrefixAndSuffixText(entry, originalNode, checker)) };
     }
 
-    export function toReferenceEntry(entry: Entry, symbol: Symbol | undefined): ReferenceEntry {
+    export function toReferenceEntry(entry: Entry, adjustedStartingNode?: Node): ReferenceEntry {
         const documentSpan = entryToDocumentSpan(entry);
         if (entry.kind === EntryKind.Span) {
-            return { ...documentSpan, isWriteAccess: false, isDefinition: false };
+            const isStartingPoint =
+                adjustedStartingNode &&
+                adjustedStartingNode.kind !== SyntaxKind.SourceFile &&
+                adjustedStartingNode.pos <= documentSpan.textSpan.start &&
+                documentSpan.textSpan.start + documentSpan.textSpan.length <= adjustedStartingNode.end &&
+                documentSpan.fileName === adjustedStartingNode.getSourceFile().fileName;
+            return {
+                ...documentSpan,
+                isWriteAccess: false,
+                isDefinition: false,
+                isStartingPoint: isStartingPoint ? true : undefined,
+            };
         }
         const { kind, node } = entry;
         return {
             ...documentSpan,
             isWriteAccess: isWriteAccessForReference(node),
-            isDefinition: isDeclarationOfSymbol(node, symbol),
+            isDefinition: isDefinitionForReference(node),
             isInString: kind === EntryKind.StringLiteral ? true : undefined,
+            isStartingPoint: (adjustedStartingNode && (adjustedStartingNode === node || adjustedStartingNode === node?.parent || adjustedStartingNode === node?.parent?.parent)) || undefined,

```

Comment:
Arguably, we should search the whole ancestry tree.  This seemed like a reasonable trade-off between correctness and performance since I couldn't find any cases that needed to go beyond the immediate parent.